sphinx-docs/screenshot.py

```diff
@@ -0,0 +1,296 @@
+    command_args = command.split()
+    if not command_args:
+        return None
+    else:
+        selector = command_args[0]
+        action = command_args[1]
+        options = command_args[2:]
+
+    if action not in ('click', 'send_keys', 'submit', ''):
+        raise ActionError(action)
+
+    # Validate input options for the specified action.
+    if action == 'click' or action == 'submit':
+        if options:
+            raise OptionError("The action '%s' must not contain any arguments whereas supplied arguments: '%s'." % (action, repr(options)))
+
+    return {'selector': selector, 'action': action, 'options': options}
+
+def _parse_login(username, password, submit=""):
+    """" Parses a LOGIN command.
+
+    Returns a dictionary with following keys:
+       runhandler (string):  "_login_handler".
+       args (dict) : A dictionary of arguments with following keys:
+         username (string):  the username.
+         password (string): password.
+         submit (bool): True if login form is to be submitted, false otherwise.
+    """
+    submit_bool = True if submit == "submit" else False
+    args = {'username': username, 'password': password, 'submit': submit_bool}
+    return {'runhandler': '_login_handler', 'args': args}
+
+def _parse_nav_steps(arg_str):
+    """ Here's how to specify the navigation steps:
+
+        1. selector action [options] ["|" selector action [options]] ...
+        2. aliased_action_sequence [options]
+
+        selector will identify the element... for now we'll just implement
+            selection by id e.g. "#username-field"
+            "NEXT", which just sends a tab keystroke
+            "SAME", which just stays focused on the element from the last action
+        Where action could be one of "click", "send_keys", or "submit":
+            click has no options and just clicks the element
+            send_keys sends a sequence of keystrokes specified as a string by options
+                (potentially with special characters representing tab, enter, etc.)
+            submit submits a form and has no options
+        Multiple actions on a page can be specified by putting a | separator,
+            and specifying the action using the same syntax.
+
+        aliased_action_sequence is one of a reserved keyword which aliases a common
+            sequence of actions as above (or performs special actions unavailable by
+            the regular syntax) potentially with options. Available aliases:
+
+            LOGIN username password [submit], which just navigates to the login page
+                and inputs the given username and password. Submits the form is submit
+                is present, otherwise not.
+
+        Returns a dictionary with:
+            runhandler: reference to function invoked in the run method
+            args:       a dictionary of arguments passed to the runhandler function
+    """
+    # The alias with its parse function
+    ALIASES = [("LOGIN", _parse_login)]
```

Comment:
Just to confirm, this is where we can insert shortcuts (aka aliases) in the future?